mm2src/mm2_main/tests/mm2_tests/tron_tests.rs

```diff
@@ -0,0 +1,512 @@
+    }
+
+    // Index 7 is funded
+    assert_eq!(account0.addresses[7].address, "TBic1drXQNM1BiBevg751GsZtv59GWb6ZK");
+    let spendable7 = &account0.addresses[7].balance.get("TRX").unwrap().spendable;
+    assert!(
+        *spendable7 > 3.into(),
+        "Expected index 7 to be funded (~5 TRX), got {:?}",
+        spendable7
+    );
+
+    block_on(mm.stop()).unwrap();
+}
+
+/// Test HD scanning detects addresses with transaction history but zero balance.
+/// Uses TRON_USED_ZERO_BALANCE_PASSPHRASE:
+/// - Index 0: funded (~1777.8 TRX)
+/// - Index 2: has tx history but 0 balance (used but empty)
+#[test]
+#[cfg(not(target_arch = "wasm32"))]
+fn test_trx_hd_scanning_detects_used_but_zero_balance_address() {
+    let coins = serde_json::json!([trx_conf()]);
+    let conf = Mm2TestConf::seednode_with_hd_account(TRON_USED_ZERO_BALANCE_PASSPHRASE, &coins);
+    let mm = block_on(MarketMakerIt::start_async(conf.conf, conf.rpc_password, None)).unwrap();
+
+    let result = block_on(task_enable_trx_result(
+        &mm,
+        TRON_NILE_NODES,
+        60,
+        Some(HDAccountAddressId {
+            account_id: 0,
+            chain: Bip44Chain::External,
+            address_id: 2,
+        }),
+    ))
+    .expect("Expected TRX HD activation to succeed");
+
+    let hd = match result {
+        EthWithTokensActivationResult::HD(hd) => hd,
+        _ => panic!("Expected HD activation result"),
+    };
+
+    let balance = match hd.wallet_balance {
+        EnableCoinBalanceMap::HD(hd_bal) => hd_bal,
+        _ => panic!("Expected EnableCoinBalanceMap::HD"),
+    };
+
+    let account0 = balance.accounts.first().expect("Expected first HD account entry");
+    assert!(
+        account0.addresses.len() >= 3,
+        "Expected at least 3 addresses (0, 1, 2), got {}",
+        account0.addresses.len()
+    );
+
+    // Index 0 should be funded
+    assert_eq!(
+        account0.addresses[0].address, "TSqB9tqfaQ1DYSdMCbVSLPzQsaNVjeu9hq",
+        "Unexpected address at index 0"
+    );
+    TronAddress::from_base58(&account0.addresses[0].address).expect("Invalid TRON address at index 0");
+    let spendable0 = &account0.addresses[0].balance.get("TRX").unwrap().spendable;
+    assert!(
+        *spendable0 > 1700.into(),
+        "Expected index 0 to have ~1777.8 TRX, got {:?}",
+        spendable0
+    );
+
+    // Index 2 should be detected (has tx history) but have zero balance
+    assert_eq!(
+        account0.addresses[2].address, "TPoJwueR4xfZCXuQTYqem4edQgoM3uV78n",
+        "Unexpected address at index 2"
+    );
+    TronAddress::from_base58(&account0.addresses[2].address).expect("Invalid TRON address at index 2");
+
+    // Verify index 2 has zero or negligible balance
+    if let Some(trx_balance) = account0.addresses[2].balance.get("TRX") {
+        assert!(
+            trx_balance.spendable < 1.into(),
+            "Expected index 2 to have ~0 TRX (used but empty), got {:?}",
+            trx_balance.spendable
+        );
```

Comment:
tbh, we must check that it has strictly zero balance, otherwise the test aint testin xD